src/rdkafka_metadata_cache.c

```diff
@@ -247,7 +287,7 @@ static struct rd_kafka_metadata_cache_entry *rd_kafka_metadata_cache_insert(
     rd_bool_t include_racks,
     rd_kafka_metadata_broker_internal_t *brokers_internal,
     size_t broker_cnt) {
-        struct rd_kafka_metadata_cache_entry *rkmce, *old;
+        struct rd_kafka_metadata_cache_entry *rkmce, *old, *old_by_id = NULL;
         rd_tmpabuf_t tbuf;
```

Comment:
Shouldn't we allocate memory in `tbuf` for topic_id as well?

Reply:
It's allocated in `rd_tmpabuf_add_alloc(&tbuf, sizeof(*rkmce));` as it's an embedded struct